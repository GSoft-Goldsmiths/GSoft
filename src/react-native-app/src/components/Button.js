import React from 'react';
import { style } from '../styles/variables';
import { Text, TouchableHighlight } from 'react-native';

export default ({ text, handler }) => (
  <TouchableHighlight
    style={style.button}
    onPress={() => handler()}
  >
    <Text style={{ color: '#fff' }}>{text}</Text>
  </TouchableHighlight>
);