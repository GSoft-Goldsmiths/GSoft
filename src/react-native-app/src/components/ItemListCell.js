import React from 'react';
import { Dimensions, Image, Text, TouchableOpacity, View } from 'react-native';
import { color, mainFontBold } from '../styles/variables';

export const ItemListCell = ({
                               name,
                               cellPressHandler,
                               id,
                               imageSource,
                               locations,
                               index,
                             }) => {
  return (
    <View style={{ paddingHorizontal: 10 }}>
      <TouchableOpacity
        onPress={() => cellPressHandler(name, id, locations, imageSource,
          index)}
        style={style.container}
      >
        <View style={style.textContainer}>
          <Text style={style.text}>{name}</Text>
        </View>
        <View style={style.imageContainer}>
          <Image
            resizeMode="cover"
            style={style.image}
            source={{ uri: imageSource, isStatic: true }}
          />
        </View>
      </TouchableOpacity>
    </View>
  );
};

const { width, height } = Dimensions.get('window');
const cellHeight = height / 2;
const cellWidth = width * 0.8;

const style = {
  container: {
    height: cellHeight,
    marginVertical: 20,
    borderRadius: 25,
    flex: 1,
    backgroundColor: color.background,
    flexDirection: 'column-reverse',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 11 },
    shadowOpacity: 0.5,
    shadowRadius: 5,
  },
  textContainer: {
    flex: 1,
    paddingVertical: 15,
    paddingHorizontal: 5,
  },
  text: {
    fontSize: 20,
    color: color.main,
    fontFamily: mainFontBold,
    right: 5,
    bottom: 5,
    textAlign: 'right',
  },
  imageContainer: {
    flex: 5,
    height: cellHeight,
    maxWidth: cellWidth,
    borderTopRightRadius: 25,
    borderTopLeftRadius: 25,
    overflow: 'hidden',
  },
  image: {
    height: '100%',
    maxWidth: cellWidth,
    width: cellWidth,
  },
};