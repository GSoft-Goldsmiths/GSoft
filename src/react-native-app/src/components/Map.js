import React from 'react';
import { View } from 'react-native';
import { MapView } from 'expo';
import { color } from '../styles/variables';

export default ({ markers, region }) => {

  mapMarkers = (markers || []).map((m, i) => (
    <MapView.Marker.Animated
      key={i + m.title}
      coordinate={m.coordinate}
      title={m.title || 'title'}
      description={m.description || 'desc'}
      pinColor={color.main}
    />
  ));

  return (
    <View style={styles.container}>
      <MapView
        show
        showsUserLocation
        followsUserLocation
        loadingEnabled
        region={region}
        style={styles.map}
      >
        {mapMarkers}
      </MapView>
    </View>
  );
}

const styles = {
  container: {
    backgroundColor: color.main,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    height: 200,
  },
  map: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
};