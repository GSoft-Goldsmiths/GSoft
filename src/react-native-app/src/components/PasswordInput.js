import React from 'react';
import { Image, Text, TextInput } from 'react-native';
import { color, fullPageView, mainFontBold } from '../styles/variables';

export default ({
                  value,
                  message,
                  placeholder,
                  handleTextChange,
                  handleSubmit,
                  imageSource,
                  title,
                  imageStyle,
                }) => {

  const Title = title
    ? <Text key="title" style={styles.title}>{title}</Text>
    : null;

  return [
    <Image
      key='image'
      style={{ ...fullPageView.image, ...imageStyle }}
      source={imageSource}
    />,
    Title,
    <TextInput
      key="textInput"
      ref={input => this._input = input}
      caretHidden
      autoFocus
      maxLength={4}
      keyboardType="numeric"
      style={styles.textInput}
      secureTextEntry={true}
      onChangeText={(text) => handleTextChange(text)}
      placeholder={placeholder}
      value={value}
      onSubmitEditing={(text) => handleSubmit(text)}
      onFocus={() => {console.log('It\'s focused now');}}
    />,
    <Text
      key='message'
      style={styles.message}>
      {message}
    </Text>,
  ];
}

const styles = {
  title: {
    ...fullPageView.title,
    fontSize: 20,
    fontFamily: mainFontBold,
    textAlign: 'center',
  },
  textInput: { height: 40, textAlign: 'center', fontSize: 20 },
  message: { color: color.danger, textAlign: 'center' },
};