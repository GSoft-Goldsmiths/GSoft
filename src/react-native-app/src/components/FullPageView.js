import React from 'react';
import { fullPageView, style } from '../styles/variables';
import Button from './Button';
import { Image, Text, View } from 'react-native';

export default ({
                  containerStyle,
                  imageStyle,
                  imageSource,
                  title,
                  subtitle,
                  buttonText,
                  buttonHandler,
                }) => {

  return (
    <View style={style.container}>
      <Image
        style={fullPageView.image}
        source={imageSource}
      />
      <Text style={fullPageView.title}>{title}</Text>
      {subtitle ? <Text style={fullPageView.subtitle}>{subtitle}</Text> : null}
      <Button text={buttonText} handler={buttonHandler}/>
    </View>
  );
}