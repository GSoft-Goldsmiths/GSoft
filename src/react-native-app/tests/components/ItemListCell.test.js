import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import { TouchableOpacity, Image, Text, View } from 'react-native';

import { ItemListCell } from '../../src/components/ItemListCell';

configure({ adapter: new Adapter() });

describe('<ItemListCell />', ()=>{
  
  it('renders without crashing', () => {
    const rendered = renderer.create(<ItemListCell 
      index={0}
      name="item name" 
      cellPressHandler={()=>{}} 
      id={123} 
      imageSource="source" 
      locations={[]} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
  
  it('renders correct elements', () => {
    const wrapper = shallow(<ItemListCell 
      index={0}
      name="item name" 
      cellPressHandler={()=>{}} 
      id={123} 
      imageSource="source" 
      locations={[]} />);
    expect(wrapper.find(View).exists()).toBeTruthy();
    expect(wrapper.find(View).length).toEqual(3);
    expect(wrapper.find(Image).exists()).toBeTruthy();
    expect(wrapper.find(Image).length).toEqual(1);
    expect(wrapper.find(Text).exists()).toBeTruthy();
    expect(wrapper.find(Text).length).toEqual(1);
    expect(wrapper.find(TouchableOpacity).exists()).toBeTruthy();
    expect(wrapper.find(TouchableOpacity).length).toEqual(1);
  });
});

