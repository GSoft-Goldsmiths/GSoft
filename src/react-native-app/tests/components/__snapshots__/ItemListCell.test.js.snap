// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemListCell /> renders without crashing 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 25,
        "flex": 1,
        "flexDirection": "column-reverse",
        "height": 667,
        "marginVertical": 20,
        "opacity": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 11,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingHorizontal": 5,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "bottom": 5,
            "color": "#0093A8",
            "fontFamily": "Caviar Dreams Bold",
            "fontSize": 20,
            "right": 5,
            "textAlign": "right",
          }
        }
      >
        item name
      </Text>
    </View>
    <View
      style={
        Object {
          "borderTopLeftRadius": 25,
          "borderTopRightRadius": 25,
          "flex": 5,
          "height": 667,
          "maxWidth": 600,
          "overflow": "hidden",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "isStatic": true,
            "uri": "source",
          }
        }
        style={
          Object {
            "height": "100%",
            "maxWidth": 600,
            "width": 600,
          }
        }
      />
    </View>
  </View>
</View>
`;
